appinventor/components/src/com/google/appinventor/components/runtime/MagneticFieldSensor.java

```diff
@@ -0,0 +1,161 @@
+// http://www.apache.org/licenses/LICENSE-2.0
+
+package com.google.appinventor.components.runtime;
+
+import android.hardware.Sensor;
+import android.hardware.SensorEvent;
+import android.hardware.SensorEventListener;
+import android.hardware.SensorManager;
+
+import com.google.appinventor.components.annotations.DesignerComponent;
+import com.google.appinventor.components.annotations.DesignerProperty;
+import com.google.appinventor.components.annotations.PropertyCategory;
+import com.google.appinventor.components.annotations.SimpleEvent;
+import com.google.appinventor.components.annotations.SimpleObject;
+import com.google.appinventor.components.annotations.SimpleProperty;
+import com.google.appinventor.components.common.ComponentCategory;
+import com.google.appinventor.components.common.YaVersion;
+
+@DesignerComponent(
+  category = ComponentCategory.SENSORS,
+  description = "<p>Non-visible component that measures the ambient geomagnetic field for all three physical axes (x, y, z) in Tesla https://en.wikipedia.org/wiki/Tesla_(unit). </p>",
+  iconName = "images/magneticSensor.png",
+  nonVisible = true,
+  version = YaVersion.MAGNETICFIELDSENSOR_COMPONENT_VERSION)
+@SimpleObject
+
+public class MagneticFieldSensor extends AndroidNonvisibleComponent implements SensorEventListener, Deleteable, OnPauseListener, OnResumeListener, OnStopListener, SensorComponent {
+    private double absoluteStrength;
+    private boolean enabled = true;
+    private boolean listening;
+    private Sensor magneticSensor;
+    private final SensorManager sensorManager;
+    private float xStrength;
+    private float yStrength;
+    private float zStrength;
+
+    public MagneticFieldSensor(ComponentContainer container) {
+        super(container.$form());
+        form.registerForOnResume(this);
+        form.registerForOnStop(this);
+        form.registerForOnPause(this);
+        sensorManager = (SensorManager) container.$context().getSystemService("sensor");
+        magneticSensor = sensorManager.getDefaultSensor(2);
+        startListening();
+    }
+
+    @SimpleProperty(category = PropertyCategory.BEHAVIOR)
+    public boolean Available() {
+        return sensorManager.getSensorList(2).size() > 0;
+    }
+
+    @SimpleProperty(category = PropertyCategory.BEHAVIOR)
+    public float MaximumRange() {
+        return magneticSensor.getMaximumRange();
+    }
+
+    @SimpleProperty(category = PropertyCategory.BEHAVIOR)
+    public boolean Enabled() {
+        return enabled;
+    }
+
+    @DesignerProperty(defaultValue = "True", editorType = "boolean")
+    @SimpleProperty
+    public void Enabled(boolean localEnabled) {
+        if (enabled != localEnabled) {
+            enabled = localEnabled;
+        }
+        if (enabled) {
+            startListening();
+        } else {
+            stopListening();
+        }
+    }
+
+    @SimpleEvent(description = "Indicates that the magnetic sensor data has changed. ")
```

Comment:
The event has a description, but the properties also need them. The descriptions because the rollover tooltips on the blocks.

Reply:
@barreeeiroo Descriptions for properties still need to be addressed.